tt_metal/llrt/tt_cluster.cpp

```diff
@@ -308,6 +315,124 @@ const metal_SocDescriptor &Cluster::get_soc_desc(chip_id_t chip) const {
+            if (core_desc.second.type != CoreType::HARVESTED) {
+                CoreCoord virtual_coords = this->get_virtual_coordinate_from_physical_coordinates(chip_id, core_desc.first, core_desc.second.type);
+                tt_cxy_pair virtual_core = tt_cxy_pair(chip_id, virtual_coords.x, virtual_coords.y);
+                tt_cxy_pair umd_core = this->get_soc_desc(chip_id).convert_to_umd_coordinates(tt_cxy_pair(chip_id, core_desc.first.x, core_desc.first.y));
+                this->virtual_to_umd_coord_mapping_[virtual_core] = umd_core;
+                if (core_desc.second.type == CoreType::WORKER) {
+                    this->virtual_worker_cores_[chip_id].insert(virtual_coords);
+                } else if (core_desc.second.type == CoreType::ETH) {
+                    this->virtual_eth_cores_[chip_id].insert(virtual_coords);
+                }
+            }
+        }
+    }
+}
+
+void Cluster::generate_logical_to_virtual_coord_mapping() {
+    for (auto chip_id : this->cluster_desc_->get_all_chips()) {
+        auto board_type = this->get_board_type(chip_id);
+        if (this->worker_logical_to_virtual_x_.find(board_type) != this->worker_logical_to_virtual_x_.end()) {
+            continue;
+        }
+        auto& soc_desc = this->get_soc_desc(chip_id);
+        this->worker_logical_to_virtual_x_.insert({board_type, {}});
+        this->worker_logical_to_virtual_y_.insert({board_type, {}});
+        this->eth_logical_to_virtual_.insert({board_type, {}});
+        for (auto x_coords : soc_desc.worker_log_to_routing_x) {
+            CoreCoord phys_core = soc_desc.get_physical_core_from_logical_core(CoreCoord(x_coords.first, 0), CoreType::WORKER);
+            CoreCoord virtual_coords = this->get_virtual_coordinate_from_physical_coordinates(chip_id, phys_core, CoreType::WORKER);
+            this->worker_logical_to_virtual_x_.at(board_type).insert({x_coords.first, virtual_coords.x});
+        }
+        for (auto y_coords : soc_desc.worker_log_to_routing_y) {
+            CoreCoord phys_core = soc_desc.get_physical_core_from_logical_core(CoreCoord(0, y_coords.first), CoreType::WORKER);
+            CoreCoord virtual_coords = this->get_virtual_coordinate_from_physical_coordinates(chip_id, phys_core, CoreType::WORKER);
+            this->worker_logical_to_virtual_y_.at(board_type).insert({y_coords.first, virtual_coords.y});
+        }
+        for (std::size_t log_eth_core_y = 0; log_eth_core_y < soc_desc.physical_ethernet_cores.size(); log_eth_core_y++) {
+            CoreCoord logical_eth_core = {0, log_eth_core_y};
+            CoreCoord virtual_coords = this->get_virtual_coordinate_from_physical_coordinates(chip_id, soc_desc.physical_ethernet_cores.at(log_eth_core_y), CoreType::ETH);
+            this->eth_logical_to_virtual_.at(board_type).insert({logical_eth_core, virtual_coords});
+        }
+    }
+
+}
+
+void Cluster::generate_virtual_to_profiler_flat_id_mapping() {
```

Comment:
Should we add #if defined(TRACY_ENABLE) here. We never need the flat_id outside of that define.

Reply:
done